src/runtime-rs/crates/hypervisor/src/device/driver/virtio_fs.rs

```diff
@@ -74,5 +80,92 @@ pub struct ShareFsDeviceConfig {
 
 #[derive(Debug, Clone)]
 pub struct ShareFsDevice {
+    pub device_id: String,
     pub config: ShareFsDeviceConfig,
 }
+
+impl ShareFsDevice {
+    // new creates a ShareFsDevice
+    pub fn new(device_id: &str, config: &ShareFsDeviceConfig) -> Self {
+        Self {
+            device_id: device_id.to_string(),
+            config: config.clone(),
+        }
+    }
+}
+
+#[async_trait]
+impl Device for ShareFsDevice {
+    async fn attach(&mut self, h: &dyn hypervisor) -> Result<()> {
+        h.add_device(DeviceType::ShareFs(self.clone()))
+            .await
+            .context("add share-fs device.")?;
+
+        return Ok(());
+    }
+
+    async fn detach(&mut self, _h: &dyn hypervisor) -> Result<Option<u64>> {
+        // no need to detach share-fs device
+
+        Ok(None)
+    }
+
+    async fn get_device_info(&self) -> DeviceType {
+        DeviceType::ShareFs(self.clone())
+    }
+
+    async fn increase_attach_count(&mut self) -> Result<bool> {
+        // share-fs devices will not be attached multiple times, Just return Ok(false)
+
+        Ok(false)
+    }
+
+    async fn decrease_attach_count(&mut self) -> Result<bool> {
+        // share-fs devices will not be detached multiple times, Just return Ok(false)
+
+        Ok(false)
+    }
+}
+
+impl ShareFsMountDevice {
+    // new creates a share-fs Mount
+    pub fn new(device_id: &str, config: &ShareFsMountConfig) -> Self {
+        Self {
+            device_id: device_id.to_string(),
+            config: config.clone(),
+        }
+    }
+}
+
+#[async_trait]
+impl Device for ShareFsMountDevice {
+    async fn attach(&mut self, h: &dyn hypervisor) -> Result<()> {
+        h.add_device(DeviceType::ShareFsMount(self.clone()))
+            .await
+            .context("add share-fs Mount.")?;
+
+        return Ok(());
+    }
+
+    async fn detach(&mut self, _h: &dyn hypervisor) -> Result<Option<u64>> {
+        // no need to detach virtio-fs mount
+
```

Comment:
The VMM supports umount the share-fs mount, we could implement the detach of virtio-fs mount.